### PR TITLE
PseudoModel for forecast, hofx4d and 3dvarfgat

### LINKED ISSUE #404: Use the oops generic PseudoModel
## Description
Even the "no advance" soca forecast does require the initialization of MOM6, which can be a significant overhead and is currently problematic when trying to run the oops coupled_hofx application

## Solution
Add the necessary method in State to allow the implementation of a PseudoModel in soca.

### PR DESCRIPTION
**Pull request by guillaumevernieres**
## Description
Adds the possibility to use the generic oops PseudoModel for forecast, hofx and var. 

### Issue(s) addressed
closes #404 

## Testing

How were these changes tested? **pseudo model implemented in the forecast, hofx and var applications**
What compilers / HPCs was it tested with? **gnu**
Are the changes covered by ctests? **yes**

Possible remaining issue: Not tested with sea-ice. I couldn't figure out how to read the sea-ice files dumped during a soca forecast, these would not be used with the pseudo model anyways (the sea-ice and ocean restarts from mom6/sis2/cice would instead). I could add a test with sea-ice, advancing the pseudo model but pointing to the same restart, as an illustration ... or not.

### CODE REVIEW

**Comment by travissluka** [comment]
TravisCI is failing because it is trying to use the matching `feature/pseudomodel` branch in oops, which is horribly out of date.
Go yell at whoever has left this incredibly old branch needlessly lying around, lurking in the shadows this entire time just to cause us all headaches in the future...... (ahem, @aerorahul )

**Comment by aerorahul** [comment]
> TravisCI is failing because it is trying to use the matching `feature/pseudomodel` branch in oops, which is horribly out of date.
> Go yell at whoever has left this incredibly old branch needlessly lying around, lurking in the shadows this entire time just to cause us all headaches in the future...... (ahem, @aerorahul )

@travissluka @guillaumevernieres 
The stale branch has been removed.  Travis-CI has been triggered.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -390,6 +396,11 @@ soca_add_test( NAME forecast_mom6_bgc
                TEST_DEPENDS test_soca_gridgen )
 endif()
 
+# Test PseudoModel forecast
+soca_add_test( NAME forecast_pseudo
+               EXE  soca_forecast.x
+             TEST_DEPENDS test_soca_forecast_mom6 )
```
Comment (aerorahul) [inline test/CMakeLists.txt]: please indent correctly.

**Review by aerorahul — APPROVED** [review]
minor comment regarding indentation.
code looks good.

**Comment by guillaumevernieres** [comment]
Thanks @aerorahul

**Pull request merged**